### PR TITLE
Improve the metrics for the storage

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

So far the `metering.rs` has the latencies. But other statistics from the storages are of importance.

## Proposal

Add some metrics representing the use case of the systems. Those metrics can be used subsequently in
benchmarking.

## Test Plan

The CI

## Release Plan

Not relevant.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-views/src/backends/metering.rs [hunk 1]
```diff
@@ -143,39 +319,117 @@ where
...
+            .find_keys_by_prefix_prefix_size
+            .with_label_values(&[])
+            .observe(key_prefix.len() as f64);
+        let result = self.store.find_keys_by_prefix(key_prefix).await?;
+        let mut num_keys = 0;
+        let mut keys_size = 0;
+        for key in result.iterator() {
+            num_keys += 1;
+            keys_size += key?.len();
+        }
```
Comment (andresilva91) [inline linera-views/src/backends/metering.rs:395]: You might be able to do something like this here:
```suggestion
        let keys_size: usize = result.iterator().map(|key| key?.len()).sum();
        let num_keys = result.iterator().count();
```

File: linera-views/src/backends/metering.rs [hunk 2]
```diff
@@ -143,39 +319,117 @@ where
...
+            .with_label_values(&[])
+            .observe(key_prefix.len() as f64);
+        let result = self.store.find_key_values_by_prefix(key_prefix).await?;
+        let mut num_keys = 0;
+        let mut key_values_size = 0;
+        for key_value in result.iterator() {
+            let (key, value) = key_value?;
+            num_keys += 1;
+            key_values_size += key.len() + value.len();
+        }
```
Comment (andresilva91) [inline linera-views/src/backends/metering.rs:423]:
```suggestion
        let key_values_size: usize = result.iterator().map(|key_value| key_value?.0.len() + key_value?.1.len()).sum();
        let num_keys = result.iterator().count();
```

**Review by andresilva91 — APPROVED** [review]
Some optional nits, but lgtm!

File: linera-views/src/backends/metering.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-views/src/backends/metering.rs:395]: There was a way to do it nicer
```[rust]
        let (num_keys, keys_size) = result
            .iterator()
            .map(|key| key.map(|k| k.len()))
            .collect::<Result<Vec<usize>, _>>()?
            .into_iter()
            .fold((0, 0), |(count, size), len| (count + 1, size + len));
```

File: linera-views/src/backends/metering.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-views/src/backends/metering.rs:423]: Thanks, similar as above.

**Pull request merged**